src/main/java/com/refinedmods/refinedstorage/apiimpl/network/node/GridNetworkNode.java

```diff
@@ -439,6 +439,16 @@ public void onClear(PlayerEntity player) {
                     network.getItemStorageTracker().changed(player, slot.copy());
                 }
             }
+        } else if (network == null || !network.canRun()) {
```

Comment:
I think this will dupe items if you have a pattern grid? Can you improve this whole if so it's readable?

```java
if (type == GridType.CRAFTING) {
  if (network != null && network.canRun() && network.getSecurityManager().hasPermission(Permission.INSERT, player)) {
    // stuff
   } else {
    // other stuff
   }
} else if (type == GridType.PATTERN) {
  clearMatrix();
}
```